src/core/moves/proposal/tree/RootTimeSlideProposal.cpp

```diff
@@ -0,0 +1,213 @@
+#include <cmath>
+#include <iostream>
+
+#include "DistributionNormal.h"
+#include "RootTimeSlideProposal.h"
+#include "RandomNumberFactory.h"
+#include "RandomNumberGenerator.h"
+#include "RbConstants.h"
+#include "Proposal.h"
+#include "StochasticNode.h"
+#include "TopologyNode.h"
+#include "Tree.h"
+
+namespace RevBayesCore { class DagNode; }
+
+using namespace RevBayesCore;
+
+/**
+ * Constructor
+ *
+ * Here we simply allocate and initialize the Proposal object.
+ */
+RootTimeSlideProposal::RootTimeSlideProposal( StochasticNode<Tree> *n, double d ) : Proposal(),
+variable( n ),
+delta( d )
+{
+    // tell the base class to add the node
+    addNode( variable );
+    
+}
+
+
+/**
+ * The cleanProposal function may be called to clean up memory allocations after AbstractMove
+ * decides whether to accept, reject, etc. the proposed value.
+ *
+ */
+void RootTimeSlideProposal::cleanProposal( void )
+{
+    ; // do nothing
+}
+
+/**
+ * The clone function is a convenience function to create proper copies of inherited objected.
+ * E.g. a.clone() will create a clone of the correct type even if 'a' is of derived type 'b'.
+ *
+ * \return A new copy of the proposal.
+ */
+RootTimeSlideProposal* RootTimeSlideProposal::clone( void ) const
+{
+    
+    return new RootTimeSlideProposal( *this );
+}
+
+
+/**
+ * Get Proposals' name of object
+ *
+ * \return The Proposals' name.
+ */
+const std::string& RootTimeSlideProposal::getProposalName( void ) const
+{
+    static std::string name = "RootTimeSlide";
+    
+    return name;
+}
+
+
+double RootTimeSlideProposal::getProposalTuningParameter( void ) const
+{
+    // this proposal has no tuning parameter
+    return RbConstants::Double::nan;
```

Comment:
isn't delta the tuning parameter ? that's what it says in the parameter description

Reply:
Good point. Probably lost this while copying ....